### PR TITLE
Update Dockerfile to fetch dependencies

### PR DESCRIPTION
**Pull request by kpouget**
This PR modifies the `Dockerfile` & `run` files to move the fetching of dependencies inside the image.

Fixes https://github.com/openshift-psap/ci-artifacts/issues/133

### CODE REVIEW

**Review by jmencak — APPROVED** [review]
Untested, but
/lgtm
the changes LG and make sense to me.

**Comment by kpouget** [comment]
I missed a `/usr/bin/gather` in the entrypoint ...

/test gpu-operator-e2e

for information, the reason I change `/usr/bin/gather` to `/usr/bin/gpu-operator_gather` is to avoid GPU Operator - specific things in the image, and I recently learned that `oc adm must-gather` accepts "`-- name`" to specify a custom script to run

**Comment by jmencak** [comment]
I have one suggestion to make. Why don't we keep PRs as WiP until they're ready/complete and then just remove the WiP for people to review and approve them? :) No need for /hold then.

**Comment by kpouget** [comment]
@jmencak, yes, that could make sense,

overall, my PR have 4 stages:
1. unfinished
--> I'm still working on it
2. ready for feedback
--> I've finished coding it and testing / reviewing the the CI logs, happy to fix according to feedback
3. finished, ready for final review&lgtm
4. finished and reviewed
--> I can have a last look and allow the bot to merge

and basically, 1. is `WIP` ... and the rest not fully clear!
I don't like that the bot merges the PR without me giving the final :+1: , so I used `\hold` to block it ...
and I've recently added `process/ready-for-review` to help distinguishing `2` from `3`

**Comment by kpouget** [comment]
I'm planning to prepare a `CONTRIBUTE` file where this could be explained & formalized; that can be a good place to discuss it and find a good workflow :)

**Comment by jmencak** [comment]
> @jmencak, yes, that could make sense,
> 
> overall, my PR have 4 stages:
> 
>     1. unfinished
>        --> I'm still working on it
> 
>     2. ready for feedback
>        --> I've finished coding it and testing / reviewing the the CI logs, happy to fix according to feedback
> 
>     3. finished, ready for final review&lgtm
> 
>     4. finished and reviewed
>        --> I can have a last look and allow the bot to merge
> 
> 
> and basically, 1. is `WIP` ... and the rest not fully clear!
> I don't like that the bot merges the PR without me giving the final +1 , so I used `\hold` to block it ...
> and I've recently added `process/ready-for-review` to help distinguishing `2` from `3`

Wow, so many stages. :) Basically, I typically only have WiP and sometimes I even skip that one and expect reviews once WiP is gone. Are people notified once you add the `process/ready-for-review` label? I'd personally vote for having less steps/KISS.  But I'll let others comment.

**Comment by kpouget** [comment]
test failed because of this flake, which should be addressed by https://github.com/openshift-psap/ci-artifacts/pull/134
```
TASK: gpu_operator_deploy_from_operatorhub : Get the GPU Operator OperatorHub ClusterServiceVersion name
<command> oc get ClusterServiceVersion -l operators.coreos.com/gpu-operator-certified.openshift-operators -oname -n openshift-operators
```
/test gpu-operator-e2e

**Comment by kpouget** [comment]
/hold cancel
tests are passing correctly now

File: build/Dockerfile
Comment (dagrayvid) [inline build/Dockerfile:6]:
```suggestion
      	io.k8s.description="An image for running Ansible artifacts for OpenShift PSAP CI" \
```
I think this description could be reworded to be more meaningful and clear.

File: build/Dockerfile
Comment (dagrayvid) [inline build/Dockerfile:5]:
```suggestion
LABEL 	io.k8s.display-name="OpenShift PSAP CI artifacts" \
```

**Comment by dagrayvid** [comment]
/lgtm

Thanks for the cleanup in run and the entitle.sh in the toolbox.

**Pull request merged**
